interpreter.asm

```diff
@@ -741,3 +723,48 @@ OLD_BASE = * + 1
 
 .chars_to_process
     !byte 0
+
++BACKLINK "dowords", 7 ; ( xt -- )
+    ; to be useful, nothing must be left on stack before execute
+    ; so that there is no distance between nt and the rest of the stack
+    lda LSB,x
+    sta .xt
+    lda MSB, x
+    sta .xt + 1
+    inx
+    lda _LATEST
+    sta .dowords_nt
+    lda _LATEST + 1
+    sta .dowords_nt + 1
+
+.dowords_lambda
+    dex
+    lda .dowords_nt
+    sta LSB, x
+    lda .dowords_nt + 1
+    sta MSB, x
+.xt = * + 1
+    jsr PLACEHOLDER_ADDRESS
+    inx
+    lda LSB-1, x
+    bne +
+-   rts
++   ldy #0
+    lda .dowords_nt
+    sta W
+    lda .dowords_nt + 1
+    sta W + 1
+    lda (W), y
+    beq -
+    and #STRLEN_MASK
+    clc
+    adc #3 ; guaranteed carry clear
+    adc .dowords_nt
+    sta .dowords_nt
+    lda .dowords_nt + 1
+    adc #0
+    sta .dowords_nt + 1
+    jmp .dowords_lambda
+; using a word here in case the lambda trashes Ws
+.dowords_nt
```

Comment:
what does dowords_nt mean? i suppose nt is an acronym? could just as well spell it out, we can afford the extra characters when cross-compiling ;-)